File: scripts/remove-elite-walkon.js
const fs = require('fs');

console.log('Removing walkon properties from elite conference schools...');

// Read the file
let content = fs.readFileSync('./src/data/schoolStandards.ts', 'utf8');

// Elite conferences
const eliteConferences = ['SEC', 'Big Ten', 'ACC', 'Big 12'];

// Process each elite conference
eliteConferences.forEach(conference => {
  console.log(`Processing ${conference} schools...`);
  
  // Escape conference name for regex
  const escapedConference = conference.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
  
  // Find all school objects for this conference and remove walkon properties
  const schoolPattern = new RegExp(
    `(\\{[^{}]*?conference:\\s*"${escapedConference}"[^{}]*?(?:\\{[^{}]*?\\}[^{}]*?)*?\\})`,
    'gs'
  );
  
  content = content.replace(schoolPattern, (schoolMatch) => {
    // Remove all walkon properties from this school
    const cleanedSchool = schoolMatch.replace(/,\s*walkon:\s*"[^"]*"/g, '');
    return cleanedSchool;
  });
});

// Write back the updated content
fs.writeFileSync('./src/data/schoolStandards.ts', content, 'utf8');

console.log('Successfully removed walkon properties from all elite conference schools!');

File: remove-elite-walkon.py
#!/usr/bin/env python3
import re

# Read the file
with open('./src/data/schoolStandards.ts', 'r') as f:
    content = f.read()

# Elite conferences
elite_conferences = ['SEC', 'Big Ten', 'ACC', 'Big 12']

# Process each elite conference
for conference in elite_conferences:
    print(f'Processing {conference} schools...')
    
    # Find all walkon properties in schools of this conference
    # Pattern to match: school object containing this conference
    pattern = rf'(\{{[^{{}}]*?conference:\s*"{re.escape(conference)}".*?\}})'
    
    def remove_walkon_from_school(match):
        school_obj = match.group(1)
        # Remove all walkon properties
        updated_obj = re.sub(r',\s*walkon:\s*"[^"]*"', '', school_obj)
        return updated_obj
    
    # Apply the replacement
    content = re.sub(pattern, remove_walkon_from_school, content, flags=re.DOTALL)

# Write back the updated content
with open('./src/data/schoolStandards.ts', 'w') as f:
    f.write(content)

print('Successfully removed walkon properties from elite conference schools!')